source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionSale/form/DispositionSalePurchasersSubForm.tsx

```diff
@@ -0,0 +1,108 @@
+import { FieldArray, useFormikContext } from 'formik';
+import React from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaTrash } from 'react-icons/fa';
+
+import { LinkButton, StyledRemoveLinkButton } from '@/components/common/buttons';
+import { ContactInputContainer } from '@/components/common/form/ContactInput/ContactInputContainer';
+import ContactInputView from '@/components/common/form/ContactInput/ContactInputView';
+import { PrimaryContactSelector } from '@/components/common/form/PrimaryContactSelector/PrimaryContactSelector';
+import { SectionField } from '@/components/common/Section/SectionField';
+import {
+  DispositionSaleContactModel,
+  WithSalePurchasers,
+} from '@/features/mapSideBar/disposition/models/DispositionSaleContactModel';
+import { getDeleteModalProps, useModalContext } from '@/hooks/useModalContext';
+
+export interface IDispositionSalePurchasersSubFormProps {
+  dispositionSaleId: number | null;
+}
+
+const DispositionSalePurchaserSubForm: React.FunctionComponent<
+  React.PropsWithChildren<IDispositionSalePurchasersSubFormProps>
+> = ({ dispositionSaleId }) => {
+  const { values } = useFormikContext<WithSalePurchasers>();
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  return (
+    <FieldArray
+      name="dispositionPurchasers"
+      render={arrayHelpers => (
+        <>
+          {values.dispositionPurchasers.map(
+            (purchaser: DispositionSaleContactModel, index: number) => (
+              <React.Fragment key={`purchaser-${index}`}>
```

Comment:
Please don't do this unless there is no readily available unique identifier. See React docs for rationale:

![image](https://github.com/bcgov/PSP/assets/41091511/b2f39dfb-b9ba-4986-815d-879393b95766)